packages/runtime/container-runtime/src/runningSummarizer.ts

```diff
@@ -333,6 +338,10 @@ export class RunningSummarizer implements IDisposable {
         summarizeReason: SummarizeReason,
         cancellationToken = this.cancellationToken): void
     {
+        // Reset our op counters
```

Comment:
SummarizeHeuristicData has a bunch of methods to deal with state transitions (recordAttempt, markLastAttemptAsSuccessful). I feel resetting of these counters should be done there, on completion of attempt (assuming we are moving counters and op handler there too).

The only thing that makes me pause is asynchrony. markLastAttemptAsSuccessful() is called long after we unpaused op processing, so that's too late to do so. We can move it to recordAttempt(), but it has two drawbacks - if summary fails, then this reset would be incorrect. Plus looking at where it is called, it's called too late too.

So maybe we should do it here (in a sense of point in time, not necessary place in code), but we should examine if there is any way to "undo" it if summarization failed. I.e. maybe store prior counters and add them (not restore) to current counters when/if summary fails?

BTW, I'd move such logic (or where such logic is initiated) into lockedSummaryAction() itself. As any restore should happen while we still hold a lock.